docs/src/main/asciidoc/hibernate-orm-panache.adoc

```diff
@@ -1,326 +1,752 @@
+
+.Procedure.
+
+
+. Add the Panache JPA extension dependency to your {ProductName} project `pom.xml` file:
++
+[source,java]
+----
+<dependencies>
+    <dependency>
+        <groupId>io.quarkus</groupId>
+        <artifactId>quarkus-hibernate-orm-panache</artifactId>
+    </dependency>
+----
 
+. Define entities as JPA entities. The following example defines the `Person` class as a JPA entity.
++
+[source,java]
+----
+@Entity
+public class Person {
+    @Id @GeneratedValue private Long id;
+    private String name;
+    private LocalDate birth;
+    private Status status;
+
+    public Long getId(){
+        return id;
+    }
+    public void setId(Long id){
+        this.id = id;
+    }
+    public String getName() {
+        return name;
+    }
+    public void setName(String name) {
+        this.name = name;
+    }
+    public LocalDate getBirth() {
+        return birth;
+    }
+    public void setBirth(LocalDate birth) {
+        this.birth = birth;
+    }
+    public Status getStatus() {
+        return status;
+    }
+    public void setStatus(Status status) {
+        this.status = status;
+    }
+}
+----
+. Define your repository and add instance methods. The following example defines the `PersonRepository` :
++
+[source,java]
+----
+@ApplicationScoped
+public class PersonRepository implements PanacheRepository<Person> {
+
+   public Person findByName(String name){
+       return find("name", name).firstResult();
+   }
+
+   public List<Person> findAlive(){
+       return list("status", Status.Alive);
+   }
+
+   public void deleteStefs(){
+       delete("name", "Stef");
+  }
+}
+----
+
++
+. Add the following code to the REST endpoint where you want to use entity operations so that the operations that are defined in `PanacheEntityBase` are available:
++
+[source,java]
+----
+@Inject
+PersonRepository personRepository;
+
+@GET
+public long count(){
+    return personRepository.count();
+}
+----
+
+=== Pananche common operations
+This section lists common operations that you can use with your entity or repository. The active record pattern examples use a `Person` entity created from the `PanacheEntity` class. The repository pattern examples use a regular JPA `Person` entity and the `PersonRepository` created from the
+`PanacheRepository` class.
+
+* Create a record (active record pattern and repository pattern):
++
 [source,java]
 ----
-// creating a person
 Person person = new Person();
 person.name = "Stef";
 person.birth = LocalDate.of(1910, Month.FEBRUARY, 1);
 person.status = Status.Alive;
+----
 
-// persist it
-person.persist();
+* Persist an entity:
+** Active record pattern:
++
+[source,java]
+----
+person.persist(person);
+----
 
-// note that once persisted, you don't need to explicitly save your entity: all
-// modifications are automatically persisted on transaction commit.
+** Repository pattern:
```

Comment:
I feel that listing those operations together distracts the viewer into viewing two methods of achieving the same things _after_ they've (hopefully) made up their mind about why style they prefer, and so it stops reading like an acceptably short usage sample and turns into a mind-numbing long list, no?

Reply:
OK. Separated.

Reply:
Thanks.